core/integrations/integrations.js

```diff
@@ -0,0 +1,139 @@
+            try {
+                var stat = fs.statSync(p);
+                if (!stat.isDirectory()) {
+                    throw 'Integration must be .js file or directory.';
+                }
+
+                var desc = require(path.join(p, 'package.json'));
+                if (!desc.name || !desc.main) {
+                    throw 'Not enough information provided to describe integration.';
+                }
+
+                list.push({
+                    name: desc.name.replace(/ /g, '-').toLowerCase(),
+                    start: path.join(p, desc.main)
+                });
+            } catch (e) {
+                console.error('Invalid integration installed (syntax error?).');
+                console.critical(e);
+            }
+        }
+    }
+    cachedIntegrations = list;
+	return list;
+};
+
+exports.setIntegrationConfigs = function(platform) {
+    for (var i = 0; i < selectedIntegrations.length; i++) {
+        selectedIntegrations[i].instance.platform = platform;
+        selectedIntegrations[i].instance.config = platform.config.loadOutputConfig(selectedIntegrations[i].name);
+        if (!selectedIntegrations[i].instance.config.commandPrefix) {
+            selectedIntegrations[i].instance.config.commandPrefix = platform.defaultPrefix;
+        }
+    }
+}
+
+exports.setIntegrations = function (integrations) {
+    if (selectedIntegrations) {
+        throw 'Cannot change integrations when they are already set.';
+    }
+
+    var i = 0;
+    try {
+        for (i = 0; i < integrations.length; i++) {
+            if (!global.coffeescriptLoaded && integrations[i].start.endsWith('.coffee')) {
+                require("coffee-script").register();
+                global.coffeescriptLoaded = true;
+            }
+            integrations[i].instance = require.once(integrations[i].start);
+
+            if (integrations[i].instance.use) {
+                integrations[i].instance = require.once('./hubot/robot.js').use(integrations[i].instance);
+            }
+        }
+        selectedIntegrations = integrations;
+        return true;
+    }
+    catch (e) {
+        console.critical(e);
+        console.error('Loading the output integration file \'' + integrations[i] + '\' failed.' +
+            '\n\nIf this is your file please ensure that it is syntactically correct.');
+        return false;
+    }
+};
+
+exports.startIntegrations = function (callback) {
+    if (!selectedIntegrations || selectedIntegrations.length === 0) {
+        throw 'Integrations must be set before starting';
+    }
+
+    if (started) {
+        throw 'Cannot start when already started.';
+    }
+
+    for (var i = 0; i < selectedIntegrations.length; i++) {
+        try {
+            console.write("Loading integration '" + selectedIntegrations[i].name + "'...\t");
+            selectedIntegrations[i].instance.start(callback);
+            console.info("[DONE]");
+        }
+        catch (e) {
+            console.error("[FAIL]");
```

Comment:
![Codacy](https://www.codacy.com/assets/images/favicon.png) Issue found: [Mixed double and single quotes.](https://www.codacy.com/app/theprojectshop/Kassy/file/2530928232/issues/source?bid=3282987&fileBranchId=3283570#l114)